### PR TITLE
wait for healthcheck to pass before proceeding in stack deploy

### PR DESCRIPTION
**Pull request by wjimenez5271**
thanks for the help figuring this out @joshwget. 

this patch provides the ability for `rancher-compose` to wait for a service to be in state `healthy` before proceeding to the next stack in the dependency chain (based on the use of `depends_on` key). Is only activated when a service has the label `wait_for_healthcheck` set to `true`.

### CODE REVIEW

File: rancher/service.go [hunk 1]
```diff
@@ -145,7 +146,22 @@ func (r *RancherService) up(create bool) error {
 		service, err = r.context.Client.Service.ActionActivate(service)
 		err = r.Wait(service)
 	}
-
+	for k, v := range r.serviceConfig.Labels {
```
Comment (joshwget) [inline rancher/service.go]: Can you change this to `if _, ok := r.serviceConfig.Labels["io.rancher.wait_for_healthcheck"]; ok {`?

Comment (joshwget) [reply rancher/service.go]: Also can you add `// TODO: revisit whether this is the best place to perform this check` above this block?

**Comment by joshwget** [comment]
Thanks @wjimenez5271!

File: rancher/service.go [hunk 2]
```diff
@@ -146,6 +147,21 @@ func (r *RancherService) up(create bool) error {
 		err = r.Wait(service)
 	}
 
+	// TODO: revisit whether this is the best place to perform this check
+	if _, ok := r.serviceConfig.Labels["io.rancher.wait_for_healthcheck"]; ok {
```
Comment (ibuildthecloud) [inline rancher/service.go]: can we just change this to `io.rancher.service.wait_for_healthcheck`

**Pull request merged**